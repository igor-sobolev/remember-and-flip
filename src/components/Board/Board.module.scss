.board {
  display: flex;
  flex-wrap: wrap;
  position: relative;

  &--disabled::after {
    content: '';
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background: rgba($color: #000000, $alpha: 0.01);
  }
}