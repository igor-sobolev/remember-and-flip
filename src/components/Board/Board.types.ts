import { Tile } from '../../state/game';

export interface BoardProps {
  boardWidth: number;
  board: Tile[];
  onFlip?: (index: number) => void;
  disabled?: boolean;
}
