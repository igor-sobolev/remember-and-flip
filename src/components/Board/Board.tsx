import React, { useRef, useState, useEffect, RefObject } from 'react';

import { BOARD_SIZE } from '../../constants';
import { useWindowSize } from '../../hooks/useWindowSize';

import { BoardProps } from './Board.types';
import styles from './Board.module.scss';

const Board: React.FunctionComponent<BoardProps> = () => {
  const boardWrapper: RefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
  const [tileSize, setTileSize] = useState(0);
  const [windowWidth] = useWindowSize();

  useEffect(() => {
    setTileSize(Number(boardWrapper?.current?.clientWidth) / BOARD_SIZE);
  }, [boardWrapper, windowWidth]);

  return (
    <div className={styles.board} ref={boardWrapper}>
      size: {tileSize}
    </div>
  );
};

export default Board;
