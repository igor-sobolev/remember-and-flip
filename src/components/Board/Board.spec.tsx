import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Board from './Board';

describe('given Board component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Board />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
