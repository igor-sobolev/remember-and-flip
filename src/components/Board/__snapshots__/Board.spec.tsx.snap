// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given Board component should match the snapshot 1`] = `
<div
  className="board"
>
  size: 
  0
</div>
`;
